code/modules/reagents/chemistry/machinery/chem_master.dm

```diff
@@ -444,6 +454,10 @@
 			update_appearance(UPDATE_OVERLAYS)
 			return TRUE
 
+		if ("setPillDuration")
+			pill_layers = clamp(params["duration"], 0, PILL_MAX_PRINTABLE_LAYERS)
```

Comment:
shouldn't we convert duration from text to number first?

Reply:
BYOND implicitly converts it, this works. At least I think the conversion on BYOND's and not TGUI's side, either way this works.